util/reggen/register.py

```diff
@@ -122,7 +124,8 @@ def __init__(self,
         self.tags = tags
 
         self.shadowed = shadowed
-        sounds_shadowy = self.name.lower().endswith('_shadowed')
+        pattern = r'^[a-z0-9_]*_shadowed(_[0-9]*){0,1}'
```

Comment:
Regex nit: Maybe the first `*` should be a `+`? Matching `_shadowed_1` seems like it's probably wrong. Also, the `{0,1}` could just be a `?` and maybe the group should be noncapturing (I always need to look up the syntax: apparently it's `?:`). So this maybe?
```
r'^[a-z0-9_]+_shadowed(?:_[0-9]*)?'
```

Reply:
Agreed, let's make this more concise. I also think the second `*` can be a `+`.